feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle/configurator-attribute-single-selection-bundle.component.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+import { ConfiguratorAttributeProductCardComponentOptions } from '../../product-card/configurator-attribute-product-card.component';
+import {
+  ConfiguratorAttributeQuantityComponentOptions,
+  Quantity,
+} from '../../quantity/configurator-attribute-quantity.component';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+
+@Component({
+  selector: 'cx-configurator-attribute-single-selection-bundle',
+  templateUrl:
+    './configurator-attribute-single-selection-bundle.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorAttributeSingleSelectionBundleComponent extends ConfiguratorAttributeBaseComponent {
+  loading$ = new BehaviorSubject<boolean>(false);
```

Comment:
There are many duplicate codes in this class and `ConfiguratorAttributeRadioButtonComponent`.

Reply:
Picked!

Reply:
@WeizhengSap, could you have a look at this [PR](https://github.com/SAP/spartacus/pull/11930)?

Reply:
Done! #11930